oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/ElasticsearchConnection.java

```diff
@@ -51,17 +58,49 @@
     private String host;
     private int port;
 
+    // API key credentials
+    private String apiKeyId;
+    private String apiKeySecret;
+
     private volatile RestHighLevelClient client;
 
     private AtomicBoolean isClosed = new AtomicBoolean(false);
 
+    /**
+     * Creates an {@code ElasticsearchConnection} instance with the given scheme, host address and port that requires no
+     * authentication.
+     *
+     * @param scheme the name {@code HttpHost.scheme} name
+     * @param host   the hostname (IP or DNS name)
+     * @param port   the port number
+     */
     public ElasticsearchConnection(String scheme, String host, Integer port) {
+        this(scheme, host, port, null, null);
+    }
+
+    /**
+     * Creates an {@code ElasticsearchConnection} instance with the given scheme, host address and port that support API
+     * key-based authentication.
+     *
+     * @param scheme       the name {@code HttpHost.scheme} name
+     * @param host         the hostname (IP or DNS name)
+     * @param port         the port number
+     * @param apiKeyId     the unique id of the API key
+     * @param apiKeySecret the generated API secret
+     * @see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api.html#security-api-keys">
+     * Elasticsearch Security API Keys
+     * </a>
+     */
+    public ElasticsearchConnection(String scheme, String host, Integer port, String apiKeyId, String apiKeySecret) {
         if (scheme == null || host == null || port == null) {
             throw new IllegalArgumentException();
         }
         this.scheme = scheme;
         this.host = host;
         this.port = port;
+
+        this.apiKeyId = apiKeyId;
+        this.apiKeySecret = apiKeySecret;
```

Comment:
Thought - While we are making changes here, would it make sense to make this a connection string instead ? (Like @thomasmueller mentioned) or maybe use a builder ?

This could later on be helpful in case we need to add certain additional (optional) params to the connection details(for certain use cases) , the constructor would not require a change and we would not need to change the usages in tests etc that don't need those additional params.

Reply:
Good idea. We could take 'inspiration' from the jdbc url format in ES (https://www.elastic.co/guide/en/elasticsearch/reference/master/sql-jdbc.html). I would add it with a separate PR.